outscale/data_source_outscale_nat_service.go

```diff
@@ -63,29 +63,14 @@ func dataSourceOutscaleOAPINatService() *schema.Resource {
 
 func dataSourceOutscaleOAPINatServiceRead(d *schema.ResourceData, meta interface{}) error {
 	conn := meta.(*OutscaleClient).OSCAPI
-
-	filters, filtersOk := d.GetOk("filter")
-	natGatewayID, natGatewayIDOK := d.GetOk("nat_service_id")
-
-	if !filtersOk && !natGatewayIDOK {
-		return fmt.Errorf("filters, or owner must be assigned, or nat_service_id must be provided")
-	}
-
-	params := oscgo.ReadNatServicesRequest{}
-
-	if filtersOk {
-		params.SetFilters(buildOutscaleOAPINatServiceDataSourceFilters(filters.(*schema.Set)))
+	req := oscgo.ReadNatServicesRequest{}
+	if filters, filtersOk := d.GetOk("filter"); filtersOk {
+		req.SetFilters(buildOutscaleOAPINatServiceDataSourceFilters(filters.(*schema.Set)))
 	}
```

Comment:
This is a singular data_source, therefore to avoid matching with multiple data_source user should add
```
filtersilters, filtersOk := d.GetOk("filter")
if !filtersOk {
   return fmt.Errorf("filters must be assigned")
}
```